runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContext.java

```diff
@@ -245,6 +248,16 @@ public static class DataflowExecutionStateTracker extends ExecutionStateTracker
     private final ContextActivationObserverRegistry contextActivationObserverRegistry;
     private final String workItemId;
 
+    /**
+     * Metadata on the message whose processing is currently being managed by this tracker. If no
+     * message is actively being processed, activeMessageMetadata will be null.
+     */
+    private ActiveMessageMetadata activeMessageMetadata = null;
```

Comment:
annotate with @Nullable
if threadsafety is required, consider using `AtomicReference<ActiveMessageMetadata>`

Reply:
ef56c0419a649d246e0ee00e927ee821361be943